plugins/plugin-chart-table/src/buildQuery.ts

```diff
@@ -88,21 +89,49 @@ export default function buildQuery(formData: TableChartFormData) {
     const moreProps: Partial<QueryObject> = {};
     if (formDataCopy.server_pagination) {
       const rowLimit = formDataCopy.extra_form_data?.custom_form_data?.row_limit;
-      // 1 - means all data
-      if (rowLimit !== 1) {
-        moreProps.row_limit = rowLimit ?? formDataCopy.server_page_length + 1; // +1 to determine if exists next page
-      }
+      moreProps.row_limit = rowLimit ?? formDataCopy.server_page_length;
       moreProps.row_offset = formDataCopy?.extra_form_data?.custom_form_data?.row_offset ?? 0;
     }
 
-    return [
-      {
-        ...baseQueryObject,
-        orderby,
-        metrics,
-        post_processing: postProcessing,
-        ...moreProps,
-      },
-    ];
+    let queryObject = {
+      ...baseQueryObject,
+      orderby,
+      metrics,
+      post_processing: postProcessing,
+      ...moreProps,
+    };
+
+    if (
+      formData.server_pagination &&
+      formData?.cachedChanges?.[formData.slice_id] &&
+      JSON.stringify(formData?.cachedChanges?.[formData.slice_id]) !==
+        JSON.stringify(queryObject.filters)
+    ) {
+      queryObject = { ...queryObject, row_offset: 0 };
+      updateExternalFormData(formData.setDataMask, 0, queryObject.row_limit ?? 0);
+    }
+    // Because we use same buildQuery for all table on the page we need split them by id
+    formData.setCachedChanges({ [formData.slice_id]: queryObject.filters });
+
+    if (formData.server_pagination) {
+      return [
+        { ...queryObject },
+        { ...queryObject, row_limit: 0, row_offset: 0, post_processing: [], is_rowcount: true },
+      ];
+    }
```

Comment:
Pretty amazing to see this in action 🚀  Now if we can only get the third object in with column totals! 🥇

Reply:
👍